pkg/config/env_var.go

```diff
@@ -84,15 +84,20 @@ func NewEnvVarListFromSlice(envList []string) (EnvVarList, error) {
 
 // RemoveEnvVarsFromList removes the env variables based on the keys provided
 // and returns a new EnvVarList
-func RemoveEnvVarsFromList(envVarList EnvVarList, keys []string) EnvVarList {
+func RemoveEnvVarsFromList(envVarList EnvVarList, keys []string) (EnvVarList, error) {
+	envVarInList := false
 	newEnvVarList := EnvVarList{}
 	for _, envVar := range envVarList {
 		// if the env is in the keys we skip it
 		if util.In(keys, envVar.Name) {
+			envVarInList = true
 			continue
 		}
 
 		newEnvVarList = append(newEnvVarList, envVar)
 	}
-	return newEnvVarList
```

Comment:
Check this line, before this function returned a EnvVarList for everycase. Now there is two cases. One where it returns the list and nil error. And another where it returns nil and error